sympy/core/numbers.py

```diff
@@ -1017,12 +1017,9 @@ class Float(Number):
     >>> _.is_Float
     False
     """
-    __slots__ = ['_mpf_', '_prec']
+    __slots__ = ['_mpf_', '_prec', 'q']
 
-    # A Float represents many real numbers,
-    # both rational and irrational.
-    is_rational = None
-    is_irrational = None
+    is_rational = True
```

Comment:
I recall there being a long discussion about this at some point, and we agreed that None should be used. Probably blaming the code will lead you to the relevant issue.

Reply:
Perhaps issue #9636 fixed/discussed in #9670. I looked at that but think the reply [below](https://github.com/sympy/sympy/pull/16916#issuecomment-497046929) applies: there is nothing different in terms of mathematical properties between Float(1) and 1, only in how they behave in a computation.

Reply:
That was my suggestion. (And I still feel that is how it should be.)

Reply:
We already know that "equal" numbers do not necessarily behave in the same way. For example if `x` is a Float and `y` is Rational then `x == y` does not imply `x + a == y + a` for all numbers `a`.

Reply:
Here is an example where the Float/float contains information about its history:

```
>>> 1 + 1e-20
1.0
>>> _ == 1
True
```
It's true but the fact that it's a float tells us that we *may* have been a truncated result and the expression that led us to that result may not be 1. Float behaves the same:
```python
>>> S(1)+1e-20
1.00000000000000
>>> _==1
True
```
If it is this uncertainty in history that we want to represent with Float, then we should disable `Float(1) == 1` from being True. A Float, then, is a way of using a single value to represent an interval of arbitrarily small width containing (as I think you said) and infinite number of Rationals (and irrationals?)